core/dbt/parser/models.py

```diff
@@ -28,10 +28,26 @@
 
 # New for Python models :p
 import ast
+
 from dbt.dataclass_schema import ValidationError
 from dbt.exceptions import ParsingException, validator_error_message, UndefinedMacroException
 
 
+class PythonValidationVisitor(ast.NodeVisitor):
+    def __init__(self):
+        super().__init__()
+        self.dbt_errors = []
+
+    def visit_FunctionDef(self, node: ast.FunctionDef) -> None:
+        if node.name == "model":
+            if not node.args.args[0].arg == "dbt":
+                self.dbt_errors.append("'dbt' not provided for model as the first argument")
+            if len(node.args.args) != 2:
+                self.dbt_errors.append(
+                    "model function should have two args, `dbt` and a session to current warehouse"
+                )
```

Comment:
👍 This is how we enforce consistency — I like it!

Even if it's not strictly necessary for a given warehouse. If users really don't like passing in unused arguments, they could do something like:
```python
def model(dbt, _):
    ...
```